doc/nrf/ug_thread_architectures.rst

```diff
@@ -204,5 +207,173 @@ It also has the following disadvantages:
 .. note::
     |connection_options_limited|
 
+.. _ug_thread_architectures_communication:
+
+NCP/RCP communication details
+=============================
+
+The NCP/RCP transport architectures include a transmit (TX) buffer that stores all the data that are to be received by the host using the Spinel protocol.
+
+.. _ug_thread_architectures_communication_priorities:
+
+NCP/RCP prioritization
+----------------------
+
+Since the Spinel protocol does not enforce any prioritization for writing data, the OpenThread NCP and RCP architectures introduce a data prioritization of their own:
+
+* High priority -- for data in the TX buffer that must be written, including data that must be written as fast as possible.
+* Low priority -- for data in the TX buffer that can be delayed or can be dropped if a high priority message is awaiting to be written.
+
+When the buffer is full, some of the low priority frames cannot be dropped and are delayed for later transmission.
+This happens for example with the @ref thread_update_commands "Unsolicited update commands", where the low priority frames are themselves prioritized in the following order:
+
+* Frames that can be delayed for later transmission ("delayable frames").
+* Frames that cannot be delayed and are dropped when the TX buffer is full ("droppable frames").
+
+.. _ug_thread_architectures_communication_rxtx:
+
+Receiving and transmitting data
+-------------------------------
+
+The Spinel communication is based on commands and responses.
+The host sends commands to NCP/RCP, and expects a response to each of them.
+
+The commands and responses are tied together with the Transaction Identifier value (TID value) in the Spinel frame header.
+Responses have a non-zero TID value, and OpenThread NCP/RCP always gives them high priority.
+
+The pending responses that do not fit into the TX buffer are queued for later execution.
+The queue is itself a buffer located above the TX buffer.
+If it is full or contains any pending responses, sending of the delayable frames is postponed and all other low priority data is dropped.
+
+Moreover, the Spinel allows sending unsolicited update commands from NCP to the host.
+See :ref:`ug_thread_architectures_communication_rxtx_tx` for details.
+
```

Comment:
When using NCP with Zephyr there is also possibility that NCP will transmit raw data, without encoding it into Spinel frames. This happens when some critical errors occurs in Zephyr and system wants to give user as many information about failure as it's possible without using interrupts. It applies mainly to log messages and is done by turning off UART interrupts and flushing everything from TX buffer without encoding it.